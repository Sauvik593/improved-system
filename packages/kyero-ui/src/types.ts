import type { LinkHTMLAttributes } from 'react';

export type BaseLinkProps =
  | {
      to: string;
    }
  | {
      to: {
        search?: string;
        pathname?: string;
      };
    };

export type LinkProps = BaseLinkProps & LinkHTMLAttributes<HTMLAnchorElement>;

export type LinkComponentProps = {
  className: string;
  children: React.ReactNode;
} & LinkProps;
