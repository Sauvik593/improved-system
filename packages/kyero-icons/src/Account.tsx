export const Account = () => {
  return (
    <svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M5.625 8.125C5.625 9.28532 6.08594 10.3981 6.90641 11.2186C7.72688 12.0391 8.83968 12.5 10 12.5C11.1603 12.5 12.2731 12.0391 13.0936 11.2186C13.9141 10.3981 14.375 9.28532 14.375 8.125C14.375 6.96468 13.9141 5.85188 13.0936 5.03141C12.2731 4.21094 11.1603 3.75 10 3.75C8.83968 3.75 7.72688 4.21094 6.90641 5.03141C6.08594 5.85188 5.625 6.96468 5.625 8.125V8.125Z"
        stroke="#1F4DEF"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M15.7609 17.3968C15.0061 16.6372 14.1085 16.0344 13.1199 15.6231C12.1312 15.2118 11.0709 15 10.0001 15C8.92926 15 7.869 15.2118 6.88032 15.6231C5.89165 16.0344 4.99408 16.6372 4.23926 17.3968"
        stroke="#1F4DEF"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M0.625 10C0.625 12.4864 1.61272 14.871 3.37087 16.6291C5.12903 18.3873 7.5136 19.375 10 19.375C12.4864 19.375 14.871 18.3873 16.6291 16.6291C18.3873 14.871 19.375 12.4864 19.375 10C19.375 7.5136 18.3873 5.12903 16.6291 3.37087C14.871 1.61272 12.4864 0.625 10 0.625C7.5136 0.625 5.12903 1.61272 3.37087 3.37087C1.61272 5.12903 0.625 7.5136 0.625 10V10Z"
        stroke="#1F4DEF"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
};
