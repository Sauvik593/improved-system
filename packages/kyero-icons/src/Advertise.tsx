export const Advertise = () => {
  return (
    <svg
      width="20"
      height="20"
      viewBox="0 0 20 20"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      aria-hidden="true"
    >
      <path
        d="M8.93115 12.8125C7.68115 11.5625 7.68115 10.2125 8.93115 8.98126C10.1812 7.75001 11.5312 7.73126 12.7624 8.98126"
        stroke="white"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      ></path>
      <path
        d="M8.93115 7.70605L11.4874 10.2561"
        stroke="white"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      ></path>
      <path
        d="M7.65625 8.98096L9.575 10.8935"
        stroke="white"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      ></path>
      <path
        d="M13.9 6.80635C14.3971 6.80635 14.8 6.4034 14.8 5.90635C14.8 5.40929 14.3971 5.00635 13.9 5.00635C13.4029 5.00635 13 5.40929 13 5.90635C13 6.4034 13.4029 6.80635 13.9 6.80635Z"
        stroke="white"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      ></path>
      <path
        d="M16.0625 2.96877H10.2062C10.0739 2.96957 9.94319 2.99712 9.82182 3.04975C9.70045 3.10238 9.59098 3.179 9.49998 3.27502L3.24998 9.94377C3.16106 10.0376 3.09209 10.1485 3.04723 10.2697C3.00237 10.3909 2.98255 10.52 2.98898 10.6491C2.9954 10.7782 3.02794 10.9047 3.08461 11.0208C3.14129 11.137 3.22093 11.2405 3.31873 11.325L9.56873 16.7938C9.75799 16.9586 10.0038 17.0436 10.2545 17.0307C10.5051 17.0179 10.741 16.9083 10.9125 16.725L16.7687 10.475C16.9374 10.2982 17.0314 10.0631 17.0312 9.81877V3.93127C17.0312 3.80435 17.0061 3.67867 16.9574 3.56149C16.9086 3.4443 16.8372 3.33792 16.7471 3.24846C16.6571 3.159 16.5502 3.08824 16.4327 3.04023C16.3152 2.99223 16.1894 2.96795 16.0625 2.96877V2.96877Z"
        stroke="white"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      ></path>
    </svg>
  );
};
