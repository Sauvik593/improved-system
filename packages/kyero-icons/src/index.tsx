export * from './AnimatedSpinner';
export * from './ArrowUp';
export * from './Account';
export * from './ArrowDown';
export * from './Advertise';
export * from './ArrowBackward';
export * from './ArrowForward';
export * from './AlertInfo';
export * from './Avatar';
export * from './BoostRocket';
export * from './Camera';
export * from './Cart';
export * from './Close';
export * from './Chat';
export * from './Checked';
export * from './ChevronDown';
export * from './ChevronLeft';
export * from './Dashboard';
export * from './DeleteIcon';
export * from './Edit';
export * from './Email';
export * from './Eye';
export * from './Floorplans';
export * from './Forbidden';
export * from './Furniture';
export * from './Gate';
export * from './Globe';
export * from './Golf';
export * from './Hamburger';
export * from './HouseIcon';
export * from './JoinLogo';
export * from './List';
export * from './Logout';
export * from './LogoWide';
export * from './Imports';
export * from './InfoIcon';
export * from './Kyero';
export * from './NextIcon';
export * from './Outside';
export * from './Parking';
export * from './Photo';
export * from './Plus';
export * from './Pool';
export * from './PrevIcon';
export * from './Processing';
export * from './Properties';
export * from './Quote';
export * from './RemoteViewing';
export * from './Reply';
export * from './School';
export * from './Search';
export * from './ShareIcon';
export * from './Star';
export * from './SunMark';
export * from './Temperature';
export * from './Tick';
export * from './Town';
export * from './QuestionMark';
export * from './WavyArrow';
export * from './Views';
export * from './VirtualTour';
export * from './XIcon';
