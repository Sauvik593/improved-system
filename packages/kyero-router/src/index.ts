import { DocsRoutes } from './docs-routes';
import { GuidesRoutes } from './guides-routes';
import { IDRoutes } from './id-routes';
import { JoinRoutes } from './join-routes';
import { SupportRoutes } from './support-routes';
import { DataRoutes } from './data-routes';
import { returnCountryRouteForLocale } from './helpers';

import { preloadRoutesFromAPI } from './preload-routes-from-api';

import type {
  RoutesFromFrontendAPI,
  Countries,
  Locales,
  CountryPreloadedRoute,
  ExternalRoutes,
  FrontendRoutes,
  SocialRoutes,
} from './types';

export const Routes = {
  id: IDRoutes,
  join: JoinRoutes,
  guides: GuidesRoutes,
  docs: DocsRoutes,
  data: DataRoutes,
  support: SupportRoutes,
};

export class KyeroRouter {
  preloadedRoutes: RoutesFromFrontendAPI | null;

  constructor() {
    this.preloadedRoutes = null;
  }

  async preloadRoutes() {
    this.preloadedRoutes = await preloadRoutesFromAPI();
  }

  getExternalRoutes(locale: string) {
    return {
      privacy: Routes.docs.privacyURL(locale),
      terms: Routes.docs.termsURL(locale),
      cookies: Routes.docs.cookiesURL(locale),
      buyersGuide: Routes.guides.buyersGuideURL(locale),
      spainGuide: Routes.guides.spainGuideURL(locale),
      portugalGuide: Routes.guides.portugalGuideURL(locale),
      franceGuide: Routes.guides.franceGuideURL(locale),
      italyGuide: Routes.guides.italyGuideURL(locale),
      pdfGuide: Routes.guides.pdfGuideURL(locale),
      podcasts: Routes.guides.podcastsURL(),
      support: Routes.support.url(),
      agentsFAQ: Routes.support.agentFAQURL(locale),
      generalFAQ: Routes.support.generalFAQURL(locale),
      buyersFAQ: Routes.support.buyersFAQURL(locale),
      join: Routes.join.joinURL(locale),
      joinContact: Routes.join.contactURL(locale),
      marketData: Routes.data.marketData(locale),
      affordability: Routes.data.affordabilityCalculator(locale),
      jobs: 'https://jobs.kyero.com',
      account: Routes.id.settingsURL(locale),
      forgotPassword: Routes.id.passwordResetURL(locale),
    };
  }

  getSocialRoutes() {
    return {
      facebook: 'https://www.facebook.com/kyero.co.uk/',
      twitter: 'https://twitter.com/kyero',
      instagram: 'https://www.instagram.com/kyero_property/',
      linkedin: 'https://www.linkedin.com/company/kyero-com/',
      youtube: 'https://www.youtube.com/channel/UCmgKXDJ0fvVzMAAp4YlMYxA',
      pinterest: 'https://www.pinterest.co.uk/kyeroproperty/',
    };
  }

  getFrontendRoutes(locale: string) {
    const { preloadedRoutes } = this;

    if (!preloadedRoutes) {
      throw new Error('Routes not preloaded');
    }

    return {
      properties_for_sale: returnCountryRouteForLocale(preloadedRoutes.properties_for_sale, locale),
      agents_search: returnCountryRouteForLocale(preloadedRoutes.agents_search, locale),
      properties_to_rent: returnCountryRouteForLocale(preloadedRoutes.properties_to_rent, locale),
      cookie_preferences: preloadedRoutes.cookie_preferences[locale as Locales],
      sitemap: preloadedRoutes.sitemap[locale as Locales],
    };
  }
}

export type {
  Countries,
  Locales,
  CountryPreloadedRoute,
  ExternalRoutes,
  FrontendRoutes,
  SocialRoutes,
};
