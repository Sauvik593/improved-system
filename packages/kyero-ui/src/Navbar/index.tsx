export * from './Navbar';
export * from './NavElement';
