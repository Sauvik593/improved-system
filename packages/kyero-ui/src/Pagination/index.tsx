import cn from 'classnames';
import { Page } from './Page';
import { Button } from './Button';
import { usePagination } from './usePagination';

import type { LinkComponentProps } from '../types';

export interface PaginationProps {
  totalPages: number;
  currentPage: number;
  pathname: string;
  nextPage: number | null;
  prevPage: number | null;
  LinkComponent?: (props: LinkComponentProps) => React.ReactElement;
  search: string;
}

const DefaultComponent = ({ to, children, className }: LinkComponentProps) => {
  const path = typeof to === 'string' ? to : `${to?.pathname}?${to?.search}`;
  return (
    <a href={path} className={cn(className)}>
      {children}
    </a>
  );
};

export const Pagination = ({
  currentPage,
  totalPages,
  pathname,
  prevPage,
  nextPage,
  LinkComponent = DefaultComponent,
  search,
}: PaginationProps) => {
  const { showPrevDots, showNextDots, pagesArray } = usePagination(search, currentPage, totalPages);
  return (
    <nav className="mt-4">
      <ul className="flex items-center gap-2">
        <Button
          page={prevPage}
          pathname={pathname}
          search={search}
          type="prev"
          LinkComponent={LinkComponent}
          testId="pagination-prev"
        />
        {totalPages >= 2 && (
          <Page
            page={1}
            currentPage={currentPage}
            pathname={pathname}
            search={search}
            LinkComponent={LinkComponent}
            testId="pagination-firstpage"
          />
        )}
        {showPrevDots && (
          <li className="font-bold" data-testid="pagination-more-prev-dots">
            ...
          </li>
        )}
        {pagesArray.map((page) => {
          return (
            <Page
              key={page}
              page={page}
              currentPage={currentPage}
              pathname={pathname}
              search={search}
              LinkComponent={LinkComponent}
              testId={`pagination-page-${page}`}
            />
          );
        })}
        {showNextDots && (
          <li className="font-bold" data-testid="pagination-more-next-dots">
            ...
          </li>
        )}
        <Page
          page={totalPages}
          currentPage={currentPage}
          pathname={pathname}
          search={search}
          LinkComponent={LinkComponent}
          testId="pagination-lastpage"
        />
        <Button
          page={nextPage}
          pathname={pathname}
          search={search}
          type="next"
          testId="pagination-next"
          LinkComponent={LinkComponent}
        />
      </ul>
    </nav>
  );
};
