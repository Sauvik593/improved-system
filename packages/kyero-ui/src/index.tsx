export * from './Alert';
export * from './ArrowLink';
export * from './Badge';
export * from './Button';
export * from './Checkbox';
export * from './CheckboxList';
export * from './FiltersField';
export * from './Footer';
export * from './LanguageSwitcher';
export * from './Navbar';
export * from './Pagination';
export * from './Radio';
export * from './RadioList';
export * from './SearchInput';
export * from './Select';
export * from './Switch';
export * from './Tooltip';
