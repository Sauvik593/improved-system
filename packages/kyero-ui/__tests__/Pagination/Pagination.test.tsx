import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Pagination, PaginationProps } from '../../src';

const getProps = (props: Partial<PaginationProps> = {}) => ({
  totalPages: 10,
  currentPage: 1,
  pathname: '/',
  nextPage: 2,
  prevPage: null,
  search: '?search=test',
  ...props,
});

const renderPagination = (props: PaginationProps) => <Pagination {...props} />;

describe('<Pagination />', () => {
  describe('when there is only 1 page results', () => {
    const props = getProps({ totalPages: 1, nextPage: null });
    const pagination = renderer.create(renderPagination(props)).toJSON();
    beforeEach(() => render(renderPagination(props)));

    it('renders correctly', () => {
      expect(pagination).toMatchSnapshot();
    });

    it('should render just a first page', () => {
      const onlyPage = screen.queryByTestId('pagination-lastpage');
      const secondPage = screen.queryByTestId('pagination-page-2');
      expect(onlyPage).toBeInTheDocument();
      expect(secondPage).toBeNull();
    });

    it('should render inactive next and prev buttons', () => {
      const prevButton = screen.queryByTestId('pagination-prev-inactive');
      const nextButton = screen.queryByTestId('pagination-next-inactive');

      expect(prevButton).toBeInTheDocument();
      expect(nextButton).toBeInTheDocument();
    });

    it('should not render "more" dots in between', () => {
      const prevDots = screen.queryByTestId('pagination-more-prev-dots');
      const nextDots = screen.queryByTestId('pagination-more-next-dots');

      expect(prevDots).toBeNull();
      expect(nextDots).toBeNull();
    });
  });

  describe('when in the middle of pagination', () => {
    const props = getProps({ currentPage: 6, nextPage: 7, prevPage: 5 });
    const pagination = renderer.create(renderPagination(props)).toJSON();
    beforeEach(() => render(renderPagination(props)));

    it('renders correctly', () => {
      expect(pagination).toMatchSnapshot();
    });

    it('should render more than one page', () => {
      const currentPage = screen.queryByTestId('pagination-page-6');
      const prevPage = screen.queryByTestId('pagination-page-5');
      const nextPage = screen.queryByTestId('pagination-page-7');

      expect(currentPage).toBeInTheDocument();
      expect(prevPage).toBeInTheDocument();
      expect(nextPage).toBeInTheDocument();
    });

    it('should render active next and prev buttons', () => {
      const nextButton = screen.queryByTestId('pagination-next');
      const prevButton = screen.queryByTestId('pagination-prev');

      expect(prevButton).toBeInTheDocument();
      expect(nextButton).toBeInTheDocument();
    });

    it('should render "more" dots in between', () => {
      const prevDots = screen.queryByTestId('pagination-more-prev-dots');
      const nextDots = screen.queryByTestId('pagination-more-next-dots');

      expect(prevDots).toBeInTheDocument();
      expect(nextDots).toBeInTheDocument();
    });

    it('should still render first page', () => {
      const firstPage = screen.queryByTestId('pagination-firstpage');

      expect(firstPage).toBeInTheDocument();
    });
  });

  describe('when at the beginning of the pagination', () => {
    const props = getProps();
    const pagination = renderer.create(renderPagination(props)).toJSON();
    beforeEach(() => render(renderPagination(props)));

    it('renders correctly', () => {
      expect(pagination).toMatchSnapshot();
    });

    it('should start at first page', () => {
      const currentPage = screen.queryByTestId('pagination-firstpage');
      const nextPage = screen.queryByTestId('pagination-page-2');
      const thirdPage = screen.queryByTestId('pagination-page-3');

      expect(currentPage).toBeInTheDocument();
      expect(nextPage).toBeInTheDocument();
      expect(thirdPage).toBeInTheDocument();
    });

    it('should render active next and prev buttons', () => {
      const prevButton = screen.queryByTestId('pagination-prev-inactive');
      const nextButton = screen.queryByTestId('pagination-next');

      expect(prevButton).toBeInTheDocument();
      expect(nextButton).toBeInTheDocument();
    });

    it('should render "more" dots at the end of pagination', () => {
      const prevDots = screen.queryByTestId('pagination-more-prev-dots');
      const nextDots = screen.queryByTestId('pagination-more-next-dots');

      expect(prevDots).toBeNull();
      expect(nextDots).toBeInTheDocument();
    });
  });

  describe('when at the end of the pagination', () => {
    const props = getProps({ currentPage: 10, nextPage: null, prevPage: 9 });
    const pagination = renderer.create(renderPagination(props)).toJSON();
    beforeEach(() => render(renderPagination(props)));

    it('renders correctly', () => {
      expect(pagination).toMatchSnapshot();
    });

    it('should render end on last page', () => {
      const currentPage = screen.queryByTestId('pagination-lastpage');
      const nextPage = screen.queryByTestId('pagination-page-11');
      const prevPage = screen.queryByTestId('pagination-page-9');

      expect(currentPage).toBeInTheDocument();
      expect(nextPage).toBeNull();
      expect(prevPage).toBeInTheDocument();
    });

    it('should render active next and prev buttons', () => {
      const prevButton = screen.queryByTestId('pagination-prev');
      const nextButton = screen.queryByTestId('pagination-next-inactive');

      expect(prevButton).toBeInTheDocument();
      expect(nextButton).toBeInTheDocument();
    });

    it('should render "more" dots at the end of pagination', () => {
      const prevDots = screen.queryByTestId('pagination-more-prev-dots');
      const nextDots = screen.queryByTestId('pagination-more-next-dots');

      expect(prevDots).toBeInTheDocument();
      expect(nextDots).toBeNull();
    });
  });
});
