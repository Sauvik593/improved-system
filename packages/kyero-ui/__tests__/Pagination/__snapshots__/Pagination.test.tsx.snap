// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> when at the beginning of the pagination renders correctly 1`] = `
<nav
  className="mt-4"
>
  <ul
    className="flex items-center gap-2"
  >
    <li
      data-testid="pagination-prev-inactive"
    >
      <svg
        className="text-ocean-40"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="16"
          cy="16"
          r="15.25"
          stroke="currentColor"
          strokeWidth="1.5"
        />
        <path
          d="M17.6665 21.3333L12.3332 15.9999L17.6665 10.6666"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1.5"
        />
      </svg>
    </li>
    <li
      data-testid="pagination-firstpage"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-ocean-100 text-white"
        href="/?page=1&search=test"
      >
        1
      </a>
    </li>
    <li
      data-testid="pagination-page-2"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=2&search=test"
      >
        2
      </a>
    </li>
    <li
      data-testid="pagination-page-3"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=3&search=test"
      >
        3
      </a>
    </li>
    <li
      data-testid="pagination-page-4"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=4&search=test"
      >
        4
      </a>
    </li>
    <li
      data-testid="pagination-page-5"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=5&search=test"
      >
        5
      </a>
    </li>
    <li
      className="font-bold"
      data-testid="pagination-more-next-dots"
    >
      ...
    </li>
    <li
      data-testid="pagination-lastpage"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=10&search=test"
      >
        10
      </a>
    </li>
    <li
      data-testid="pagination-next"
    >
      <a
        className="flex h-[32px] w-[32px] items-center justify-center"
        href="/?page=2&search=test"
      >
        <svg
          className="text-ocean-100"
          fill="none"
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="16"
            cy="16"
            r="15.25"
            stroke="currentColor"
            strokeWidth="1.5"
            transform="rotate(-180 16 16)"
          />
          <path
            d="M14.3335 10.6667L19.6668 16.0001L14.3335 21.3334"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1.5"
          />
        </svg>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> when at the end of the pagination renders correctly 1`] = `
<nav
  className="mt-4"
>
  <ul
    className="flex items-center gap-2"
  >
    <li
      data-testid="pagination-prev"
    >
      <a
        className="flex h-[32px] w-[32px] items-center justify-center"
        href="/?page=9&search=test"
      >
        <svg
          className="text-ocean-100"
          fill="none"
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="16"
            cy="16"
            r="15.25"
            stroke="currentColor"
            strokeWidth="1.5"
          />
          <path
            d="M17.6665 21.3333L12.3332 15.9999L17.6665 10.6666"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1.5"
          />
        </svg>
      </a>
    </li>
    <li
      data-testid="pagination-firstpage"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=1&search=test"
      >
        1
      </a>
    </li>
    <li
      className="font-bold"
      data-testid="pagination-more-prev-dots"
    >
      ...
    </li>
    <li
      data-testid="pagination-page-6"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=6&search=test"
      >
        6
      </a>
    </li>
    <li
      data-testid="pagination-page-7"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=7&search=test"
      >
        7
      </a>
    </li>
    <li
      data-testid="pagination-page-8"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=8&search=test"
      >
        8
      </a>
    </li>
    <li
      data-testid="pagination-page-9"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=9&search=test"
      >
        9
      </a>
    </li>
    <li
      data-testid="pagination-lastpage"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-ocean-100 text-white"
        href="/?page=10&search=test"
      >
        10
      </a>
    </li>
    <li
      data-testid="pagination-next-inactive"
    >
      <svg
        className="text-ocean-40"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="16"
          cy="16"
          r="15.25"
          stroke="currentColor"
          strokeWidth="1.5"
          transform="rotate(-180 16 16)"
        />
        <path
          d="M14.3335 10.6667L19.6668 16.0001L14.3335 21.3334"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1.5"
        />
      </svg>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> when in the middle of pagination renders correctly 1`] = `
<nav
  className="mt-4"
>
  <ul
    className="flex items-center gap-2"
  >
    <li
      data-testid="pagination-prev"
    >
      <a
        className="flex h-[32px] w-[32px] items-center justify-center"
        href="/?page=5&search=test"
      >
        <svg
          className="text-ocean-100"
          fill="none"
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="16"
            cy="16"
            r="15.25"
            stroke="currentColor"
            strokeWidth="1.5"
          />
          <path
            d="M17.6665 21.3333L12.3332 15.9999L17.6665 10.6666"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1.5"
          />
        </svg>
      </a>
    </li>
    <li
      data-testid="pagination-firstpage"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=1&search=test"
      >
        1
      </a>
    </li>
    <li
      className="font-bold"
      data-testid="pagination-more-prev-dots"
    >
      ...
    </li>
    <li
      data-testid="pagination-page-4"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=4&search=test"
      >
        4
      </a>
    </li>
    <li
      data-testid="pagination-page-5"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=5&search=test"
      >
        5
      </a>
    </li>
    <li
      data-testid="pagination-page-6"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-ocean-100 text-white"
        href="/?page=6&search=test"
      >
        6
      </a>
    </li>
    <li
      data-testid="pagination-page-7"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=7&search=test"
      >
        7
      </a>
    </li>
    <li
      data-testid="pagination-page-8"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=8&search=test"
      >
        8
      </a>
    </li>
    <li
      className="font-bold"
      data-testid="pagination-more-next-dots"
    >
      ...
    </li>
    <li
      data-testid="pagination-lastpage"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-white"
        href="/?page=10&search=test"
      >
        10
      </a>
    </li>
    <li
      data-testid="pagination-next"
    >
      <a
        className="flex h-[32px] w-[32px] items-center justify-center"
        href="/?page=7&search=test"
      >
        <svg
          className="text-ocean-100"
          fill="none"
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="16"
            cy="16"
            r="15.25"
            stroke="currentColor"
            strokeWidth="1.5"
            transform="rotate(-180 16 16)"
          />
          <path
            d="M14.3335 10.6667L19.6668 16.0001L14.3335 21.3334"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1.5"
          />
        </svg>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> when there is only 1 page results renders correctly 1`] = `
<nav
  className="mt-4"
>
  <ul
    className="flex items-center gap-2"
  >
    <li
      data-testid="pagination-prev-inactive"
    >
      <svg
        className="text-ocean-40"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="16"
          cy="16"
          r="15.25"
          stroke="currentColor"
          strokeWidth="1.5"
        />
        <path
          d="M17.6665 21.3333L12.3332 15.9999L17.6665 10.6666"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1.5"
        />
      </svg>
    </li>
    <li
      data-testid="pagination-lastpage"
    >
      <a
        className="flex h-[32px] w-[32px]  items-center justify-center rounded-full bg-ocean-100 text-white"
        href="/?page=1&search=test"
      >
        1
      </a>
    </li>
    <li
      data-testid="pagination-next-inactive"
    >
      <svg
        className="text-ocean-40"
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="16"
          cy="16"
          r="15.25"
          stroke="currentColor"
          strokeWidth="1.5"
          transform="rotate(-180 16 16)"
        />
        <path
          d="M14.3335 10.6667L19.6668 16.0001L14.3335 21.3334"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1.5"
        />
      </svg>
    </li>
  </ul>
</nav>
`;
